purchase_discount/models/purchase_order.py

```diff
@@ -67,13 +67,15 @@ def _get_stock_move_price_unit(self):
         """
         price_unit = False
         price = self._get_discounted_price_unit()
-        if price != self.price_unit:
+        if price != self.price_unit and not self.env.context.get(
+            "bypass_override_price_unit"
+        ):
             # Only change value if it's different
             price_unit = self.price_unit
-            self.price_unit = price
+            self.with_context(bypass_override_price_unit=True).price_unit = price
         price = super()._get_stock_move_price_unit()
-        if price_unit:
-            self.price_unit = price_unit
+        if price_unit and not self.env.context.get("bypass_override_price_unit"):
```

Comment:
This context check could be on the beggining of the method override and simply return super() if present